example/spec/parallel/SyncGranuleTask/SyncZeroByteFileSpec.js

```diff
@@ -0,0 +1,162 @@
+
+describe('The SyncGranule task with a 0 byte file to be synced', () => {
+  let beforeAllFailed = false;
+  let collection;
+  let granuleId;
+  let prefix;
+  let provider;
+  let syncGranuleOutput;
+  let parsedFileUrl;
+
+  const filename = '0byte.dat';
+
+  beforeAll(async () => {
+    try {
+      const config = await loadConfig();
+
+      prefix = config.stackName;
+
+      const { fakeS3ProviderBucket } = await fetchFakeS3ProviderBuckets();
+
+      // Stage zero byte file for sync-granule
+      await S3.s3PutObject({
+        Bucket: fakeS3ProviderBucket,
+        Key: filename,
+        Body: '',
+      });
+
+      // Create the collection
+      collection = await createCollection(
+        prefix,
+        {
+          duplicateHandling: 'version',
+          process: 'modis',
+        }
+      );
+
+      // Create the S3 provider
+      provider = await createProvider(prefix, { host: fakeS3ProviderBucket });
+
+      granuleId = randomId('granule-id-');
+
+      const FunctionName = `${prefix}-SyncGranule`;
+      const functionConfig = await lambda().getFunctionConfiguration({
+        FunctionName,
+      }).promise();
+
+      const Payload = JSON.stringify({
+        cma: {
+          task_config: {
+            buckets: '{$.meta.buckets}',
+            provider: '{$.meta.provider}',
+            collection: '{$.meta.collection}',
+            stack: '{$.meta.stack}',
+            downloadBucket: '{$.cumulus_meta.system_bucket}',
+            duplicateHandling: '{$.meta.collection.duplicateHandling}',
+            cumulus_message: {
+              input: '{$.payload}',
+              outputs: [
+                {
+                  source: '{$}',
+                  destination: '{$.payload}',
+                },
+              ],
+            },
+          },
+          event: {
+            cumulus_meta: {
+              system_bucket: config.bucket,
+            },
+            meta: {
+              buckets: config.buckets,
+              collection,
+              provider,
+              stack: config.stackName,
+            },
+            payload: {
+              granules: [
+                {
+                  granuleId,
+                  dataType: collection.name,
+                  version: collection.version,
+                  files: [
+                    {
+                      bucket: fakeS3ProviderBucket,
+                      name: filename,
+                      path: '',
+                      size: 0,
+                    },
+                  ],
+                },
+              ],
+            },
+          },
+        },
+      });
+
+      syncGranuleOutput = await pTimeout(
+        lambda().invoke({ FunctionName, Payload }).promise(),
+        (functionConfig.Timeout + 10) * 1000
+      );
+    } catch (error) {
+      beforeAllFailed = true;
+      throw error;
+    }
+  });
+
+  it('succeeds', async () => {
+    if (beforeAllFailed) fail('beforeAll() failed');
```

Comment:
Just curious. I see this pattern in all of our integration tests. Is the idea here that we want to get to an actual test failure because if a `before` block fails it'll stop the test runner? Or not report the failure correctly?

Reply:
Yeah, Jasmine will still try to run the tests even if there is an error in `beforeAll`:

https://github.com/jasmine/jasmine/issues/1533

So the test would fail because the necessary setup in beforeAll did not complete, but it would be completely non-obvious that is what happened. To make it more obvious, we're trapping the error in beforeAll and manually failing the tests to say that is what happened.

The better solution is upgrading our integration tests to use a new test runner, since there seems to be no momentum towards fixing this in Jasmine. But that's obviously a big effort.

Reply:
Good to know! Thanks for the explanation. Also glad there's a workaround 😄